src/es-module-shims.js

```diff
@@ -57,12 +57,11 @@ Object.defineProperties(importShim, {
   l: { value: undefined, writable: true },
   e: { value: undefined, writable: true }
 });
-importShim.fetch = url => fetch(url);
-importShim.skip = /^https?:\/\/(cdn\.pika\.dev|dev\.jspm\.io|jspm\.dev)\//;
+const importShimInitialOptions = self.importShimInitialOptions || {};
+importShim.fetch = importShimInitialOptions.fetch || (url => fetch(url));
+importShim.skip = importShimInitialOptions.skip || /^https?:\/\/(cdn\.pika\.dev|dev\.jspm\.io|jspm\.dev)\//;
+importShim.onerror = importShimInitialOptions.onerror || ((e) => { throw e; });
 importShim.load = processScripts;
```

Comment:
Then maybe we could invert the control here and have a `init` hook that returns a bag with the "instance object" (no longer `importShim` itself) including the `load` function.

Reply:
Wanted to confirm we're on the same page here since I'm not sure if I've fully understood your comment. 

Did you mean asking the user to supply a init function that returns the options object instead of supplying an options object directly?

```js
// in some user supplied script before esms loads
self.esmsInit = () => {
  return {
    // whatever the user wants to supply for options
  }
}

// in esms implementation
const userSuppliedOptions = self.esmsInit ? self.esmsInit() : {}

const options = {
  fetch: userSuppliedOptions.fetch || (url => fetch(url)),
  skip: userSuppliedOptions.skip || /^https?:\/\/(cdn\.pika\.dev|dev\.jspm\.io|jspm\.dev)\//,
  onerror: userSuppliedOptions.onerror || ((e) => { throw e; }),
  load: userSuppliedOptions.load || processScripts
}

// ...

if (hasDocument) {
  options.load();
  waitingForImportMapsInterval = setInterval(options.load, 20);
}
```

Curious what the benefit would be over just asking for an object? Or maybe I'm completely off base with my interpretation of your comment? Please clarify if that's the case. Or if you feel it might be easier, feel free to push changes onto this PR directly.

Reply:
I was thinking something like:

```js
globalThis.esmsInit = {
  fetch: fetchHook,
  skip,
  onerror,
  init ({ load, ...any other calls / info / useful runtime instance state }) {
    bindLoadCall(load);
  }
};

```

Reply:
It sounds like you probably have some use case in mind for this init hook that I might be still missing.

I've pushed an update with just the suggested rename for now: https://github.com/guybedford/es-module-shims/pull/109/commits/06410f9567da40b76e418e2f4072ccef07fc033d

It might be easier at this point to get this merged in as is, so you can follow up with all the changes you had in mind for the next release without any further back and forth to get me up to speed. Wdyt?

Reply:
The `importShim` instance was always a hack, so the main suggestion was just to treat it as a function only and remove all state from that global which seems like a nice simplification your work enables.

Certainly we can treat that as a separate PR though.

Reply:
Here's my attempt at decoupling all the state from the importShim function: https://github.com/guybedford/es-module-shims/pull/109/commits/780a1f7ab53a7240aa60270d22ee5c3ca5cadaca

Basically I've moved everything that used to be on `importShim` to a separate `esmsState` object. Initially I tried to not expose it globally, but it seemed like this import.meta implementation required it: https://github.com/guybedford/es-module-shims/blob/master/src/es-module-shims.js#L122

Definitely open to other suggestions on ways to work around that.